torch/csrc/jit/passes/constant_propagation.cpp

```diff
@@ -300,6 +300,7 @@ struct ConstantPropagator {
         loop_body->eraseOutput(loop_body_offset + i);
       }
     }
+    made_change_ |= initial_outputs != node->outputs().size();
```

Comment:
Similar to above. It is better to set `made_change_ = true` when changes are made. This could be more robust if the transformation happens to alter the graph w/o changing the number of nodes.

Reply:
Same question, do you think it's realistic that `removeExtraLoopOutputs` would ever add more outputs ?

Reply:
I'd hope not :-).  But it also seems a bit more intuitive/safe to set `made_change_` at the exact point where a change is made.